### PR TITLE
Optimize fe start logic

### PR DESCRIPTION
**Pull request by morningman**
ISSUE #2051

### CODE REVIEW

**Comment by kangkaisen** [comment]
@morningman Hi. This PR effect is big, please add more test if possible.

**Comment by morningman** [comment]
> @morningman Hi. This PR effect is big, please add more test if possible.

It's OK, this CL does not change any meta date read/write format and logic, so it won't cause meta data error. 

And I am running this branch on our stability test environment. I will not merge this PR until everything is OK.

File: fe/src/main/java/org/apache/doris/catalog/Catalog.java [hunk 1]
```diff
@@ -2125,103 +2136,138 @@ private void setCanRead(boolean hasLog, boolean err) {
...
+    }
+
+    public void notifyNewFETypeTransfer(FrontendNodeType newType) {
+        try {
+            String msg = "notify new FE type transfer: " + newType;
+            LOG.warn(msg);
+            Util.stdoutWithTime(msg);
+            this.typeTransferQueue.put(newType);
+        } catch (InterruptedException e) {
+            LOG.error("failed to put new FE type: {}", newType, e);
```
[the remark this answers is not in the document]
Comment (morningman) [reply fe/src/main/java/org/apache/doris/catalog/Catalog.java]: It almost never happen. Perhaps I should exit the FE process.

**Comment by kangkaisen** [comment]
> > @morningman Hi. This PR effect is big, please add more test if possible.
> 
> It's OK, this CL does not change any meta date read/write format and logic, so it won't cause meta data error.
> 
> And I am running this branch on our stability test environment. I will not merge this PR until everything is OK.

I see. OK. Thank you.

File: fe/src/main/java/org/apache/doris/catalog/Catalog.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (kangkaisen) [reply fe/src/main/java/org/apache/doris/catalog/Catalog.java]: > It almost never happen

I see.

> Perhaps I should exit the FE process.

According to your optimization goal，I also think we should exit the FE process directly.

**Comment by morningman** [comment]
@kangkaisen Hi, do you need a further test of  this CL? My test goes well

**Comment by kangkaisen** [comment]
> @kangkaisen Hi, do you need a further test of this CL? My test goes well

I think I needn't, because I don't have special  test case for this CL and I believe you.

**Review by imay — DISMISSED** [review]
LGTM

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**